src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -1215,9 +1215,9 @@ private static bool HadLambdaConversionError(BindingDiagnosticBag diagnostics, A
             }
             else if (refArg != refParameter && !(refArg == RefKind.None && refParameter == RefKind.In))
             {
-                if (refParameter == RefKind.None || refParameter == RefKind.In)
+                if (refParameter is RefKind.None or RefKind.In or RefKind.RefReadOnlyParameter)
```

Comment:
>if (refParameter is RefKind.None or RefKind.In or RefKind.RefReadOnlyParameter) [](http://example.com/codeflow?start=16&length=79)

Something doesn't feel right here. Some of the new mismatches are allowed with a warning. Not only that, but some are allowed conditionally based on language version. However, if we get here, we end up with an error no matter what. Are we testing this code path for new mismatches?

Reply:
Good catch, thanks. The mismatches are not a problem - they are handled by `GetEffectiveParameterRefKind`, so we don't get here normally. But if there's another problem, like invalid type, overload resolution continues here and wrong error would be reported, need to extend the condition.